### PR TITLE
CEDS-2485 - Authorisations to two-step remove pattern

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/views/declaration/declaration_holder_add.scala.html [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+@(mode: Mode, form: Form[DeclarationHolder], holders: Seq[DeclarationHolder])(implicit request: JourneyRequest[_], messages: Messages)
+
+@tariffBody = { @Html(messages("declaration.type.holderOfAuthorisationSummaryText", components.details_content_link())) }
+
+@backLink = @{
+    if(holders.nonEmpty)
+        DeclarationHolderController.displayPage(mode)
+    else
+        Navigator.backLink(DeclarationHolder, mode)
```
Comment (PatrykRudnicki) [inline app/views/declaration/declaration_holder_add.scala.html:50]: Should this logic be in the Navigator? Similar like clearance requests helper methods?

File: conf/application.conf [hunk 2]
```diff
@@ -126,7 +126,7 @@ google-analytics {
 }
 
 assets {
-  version = "3.8.0"
+  version = "3.3.2"
```
Comment (PatrykRudnicki) [inline conf/application.conf:129]: Why this change?

`3.3.2` assets has problem with jQuery

File: docs/Customs Declare Exports AutoComplete.js [hunk 3]
```diff
@@ -439,6 +419,10 @@ function holderOfAuthorisation(){
                 break;
         }
     }
+    if (currentPageIs("/customs-declare-exports/declaration/authorisations-required")) {
+        document.getElementById('code_no').checked = 'checked';
+        document.getElementById('submit').click();
+    }
```
Comment (PatrykRudnicki) [inline docs/Customs Declare Exports AutoComplete.js:425]: Do we need case for the remove page?

File: conf/application.conf (hunk 2 above)
Comment (SteveSugden) [reply conf/application.conf:129]: Shouldn't have committed that change - was done in the "IE panic" changes on Friday

File: app/views/declaration/declaration_holder_add.scala.html (hunk 1 above)
Comment (SteveSugden) [reply app/views/declaration/declaration_holder_add.scala.html:50]: Possibly.  But I'd have to create a brand new `DeclarationPage` object for this. `DeclarationHolder` is the Navigator key for navigating from the summary page.  The other "helper" pages - add, remove, change - all always go back to the summary page, except this one instance (no existing holders)

File: docs/Customs Declare Exports AutoComplete.js (hunk 3 above)
Comment (SteveSugden) [reply docs/Customs Declare Exports AutoComplete.js:425]: Do we do that for other pages that are not on the direct path from start -> finish?

Comment (PatrykRudnicki) [reply docs/Customs Declare Exports AutoComplete.js:425]: I don't know, I just thought about remove page, not sure why

File: docs/Customs Declare Exports AutoComplete.js [hunk 4]
```diff
@@ -374,38 +374,18 @@ function additionalActors(){
 }
 
 function holderOfAuthorisation(){
-    if (currentPageIs("/customs-declare-exports/declaration/holder-of-authorisation")) {
+    if (currentPageIs("/customs-declare-exports/declaration/add-authorisation-required")) {
```
Comment (PatrykRudnicki) [inline docs/Customs Declare Exports AutoComplete.js:377]: Please bump the version of QS.

I always forgot about it, don't know how I can remember it during review :D

File: app/views/declaration/declaration_holder_add.scala.html (hunk 1 above)
Comment (SteveSugden) [reply app/views/declaration/declaration_holder_add.scala.html]: Created new `DeclarationPage` object and moved logic to Navigator.  This allowed me to change the view to stop passing in the existing holders collection just for the purpose of back-link creation

File: conf/application.conf (hunk 2 above)
Comment (SteveSugden) [reply conf/application.conf]: Done

File: docs/Customs Declare Exports AutoComplete.js (hunk 4 above)
Comment (SteveSugden) [reply docs/Customs Declare Exports AutoComplete.js:377]: Done
